### PR TITLE
A new way to do boundary check

### PR DESCRIPTION
**Pull request by flyingbee2012**
Fix an issue where resizing is not working properly for vertically merged cells. The fix is to resize current column and next column separately.

### CODE REVIEW

File: packages/roosterjs-editor-plugins/lib/plugins/TableResize/TableResize.ts [hunk 1]
```diff
@@ -601,7 +601,44 @@ export default class TableResize implements EditorPlugin {
         });
     };
 
+    private canResizeColums = (newPos: number): boolean => {
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-plugins/lib/plugins/TableResize/TableResize.ts:604]: typo: columns

File: packages/roosterjs-editor-plugins/lib/plugins/TableResize/TableResize.ts [hunk 2]
```diff
@@ -601,7 +601,44 @@ export default class TableResize implements EditorPlugin {
...
+        for (let i = 0; i < this.currentCellsToResize.length; i++) {
+            const td = this.currentCellsToResize[i];
+            const rect = normalizeRect(td.getBoundingClientRect());
+            if (this.isRTL && rect.right - newPos < MIN_CELL_WIDTH) {
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-plugins/lib/plugins/TableResize/TableResize.ts:608]: We still see this pattern to get width for LTR and RTL, I still think we should extract it to be a function.
Same for line 621

Comment (JiuqingSong) [reply packages/roosterjs-editor-plugins/lib/plugins/TableResize/TableResize.ts:608]: and line 626, 630

Comment (flyingbee2012) [reply packages/roosterjs-editor-plugins/lib/plugins/TableResize/TableResize.ts]: fixed

File: packages/roosterjs-editor-plugins/lib/plugins/TableResize/TableResize.ts (hunk 1 above)
Comment (flyingbee2012) [reply packages/roosterjs-editor-plugins/lib/plugins/TableResize/TableResize.ts:604]: good catch!

File: packages/roosterjs-editor-plugins/lib/plugins/TableResize/TableResize.ts (hunk 2 above)
Comment (JiuqingSong) [reply packages/roosterjs-editor-plugins/lib/plugins/TableResize/TableResize.ts]: Was this resolved? I still see it in the code.

I expect something like:
if (getDistance(rect, newPos, true /*toLeft*/, this.isRTL) <MIN_CELL_WIDTH) ...

If you think 4 parameters are too many, we can split it into two function:
function getDistanceToLeft(rect, pos, isRtl) {}
and
function getDistanceToRight(rect, pos, isRtl {}

Comment (flyingbee2012) [reply packages/roosterjs-editor-plugins/lib/plugins/TableResize/TableResize.ts]: fixed now

**Review by JiuqingSong — APPROVED** [review]

**Pull request merged**